### ISSUE TITLE
Numeric concepts not allowing decimal values

### ISSUE THREAD

**Issue by mahalakshme** [avniproject/avni-client#1007]
https://images.zenhubusercontent.com/64590403/2a5b9474-cc9c-4592-8d69-d0b4169c77fb/screen_recording_2023_06_05_at_5_10_11_pm.mov

this also will get solved: https://app.zenhub.com/workspaces/avni-product-635b73bded85e50018871ae6/issues/gh/avniproject/avni-client/968

Out of scope:
https://app.zenhub.com/workspaces/avni-product-635b73bded85e50018871ae6/issues/gh/avniproject/avni-client/1008 - since dont want to touch critical parts

Things to test:
- autopopulate numeric fields from rules
- entering decimal values and its autopopulation via rules
- readonly numeric fields
- spaces and characters in numeric fields

**Comment by petmongrels** [comment]
What was the reason the value was not accepted? Is it because the model doesn't accept it? or the action doesn't accept? Shouldn't it be handled by validation like we have regex fields?

**Comment by mahalakshme** [comment]
When typing, "1.2", there will be a time where the state value will be only "1.". At this point of time, at the model level(observations) we are doing _.toNumber -> this was erasing the decimal point. Hence we are in need to store the value as string in the NumericFormElement state to retain the dot and continue typing. 

Regarding using regex fields we can try this for consistency: This will involve changes in FormElement validate function and PrimitiveValue.js valueFromString(). Since these are coreparts and most of the testing for release 3.39 is already done, I feel we can create a story for moving the validations to regex in one of the upcoming releases.

whereas the solution I have implemented was already done by Vinay in one of the commits around 6 months back. But when doing changes for an another bug, I had by mistake broken the fix. So the solution that Vinay made(I implemented) was already tested and used by many users. So do u think its fine to proceed with this solution for now? @petmongrels
